tests/models/test_sasrec.py

```diff
@@ -582,3 +593,349 @@ def test_get_dataloader_recommend(
+
+    def test_get_dense_item_features(self, dataset_item_features: Dataset) -> None:
+        items = torch.from_numpy(
+            dataset_item_features.item_id_map.convert_to_internal(INTERACTIONS[Columns.Item].unique())
+        )
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(dataset_item_features, n_factors=5, dropout_rate=0.5)
+
+        actual_feature_dense = cat_item_embeddings.get_dense_item_features(items)
+        expected_feature_dense = torch.tensor(
+            [
+                [1.0, 0.0, 1.0, 0.0, 0.0],
+                [1.0, 0.0, 0.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0, 0.0, 0.0],
+                [1.0, 0.0, 0.0, 0.0, 1.0],
+                [0.0, 1.0, 0.0, 1.0, 0.0],
+                [0.0, 1.0, 0.0, 0.0, 1.0],
+            ],
+            device=cat_item_embeddings.device,
+        )
+
+        assert torch.equal(actual_feature_dense, expected_feature_dense)
+
+    @pytest.mark.parametrize("n_factors", (10, 100))
+    def test_create_from_dataset(self, n_factors: int, dataset_item_features: Dataset) -> None:
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5
+        )
+
+        actual_item_features = cat_item_embeddings.item_features
+        actual_n_items = cat_item_embeddings.n_items
+        actual_n_cat_features = cat_item_embeddings.n_cat_features
+        actual_embedding_dim = cat_item_embeddings.category_embeddings.embedding_dim
+
+        expected_item_features = dataset_item_features.item_features
+        # TODO: remove after adding Dense Features support
+        if isinstance(expected_item_features, SparseFeatures):
+            expected_cat_item_features = expected_item_features.get_cat_features()
+
+            assert_feature_set_equal(actual_item_features, expected_cat_item_features)
+            assert actual_n_items == dataset_item_features.item_id_map.size
+            assert actual_n_cat_features == len(expected_cat_item_features.names)
+            assert actual_embedding_dim == n_factors
+
+    @pytest.mark.parametrize(
+        "n_items,n_factors",
+        (
+            (
+                2,
+                10,
+            ),
+            (
+                4,
+                100,
+            ),
+        ),
+    )
+    def test_embedding_shape_after_model_pass(
+        self, dataset_item_features: Dataset, n_items: int, n_factors: int
+    ) -> None:
+        items = torch.from_numpy(
+            np.random.choice(dataset_item_features.item_id_map.internal_ids, size=n_items, replace=False)
+        )
+        cat_item_embeddings = IdEmbeddingsItemNet.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5
+        )
+
+        expected_item_ids = cat_item_embeddings(items)
+        assert expected_item_ids.shape == (n_items, n_factors)
+
+    def test_raises_when_dataset_no_features(self) -> None:
```

Comment:
This is not an expected behaviour (see other comment). This block (CatFeaturesItemNet) should be ignored if there are:
1. No features in dataset
2. Features are dense
3. No category features are present in sparse features
This block should just be dropped from the item net.

These would be good cases for tests. Model shouldn't raise errors in any of these cases.

Reply:
Fixed